tests/integration/components/task-card-test.js

```diff
@@ -49,7 +49,7 @@ test('it renders all the required elements', function(assert) {
   };
 
   set(this, 'task', task);
-  this.render(hbs`{{task-card task=task}}`);
+  page.render(hbs`{{task-card task=task}}`);
```

Comment:
You want to call `renderPage()` in all of this instances instead.

The general flow of an integration test is

- prepare the data and the component - set all the data and action handlers, mock all the requires servicesw, etc. 
- render the page - using `renderPage`
- simulate using the component on the page in some way - using `page`

- assert that the thing that needed to happen happened